js/reference.js

```diff
@@ -346,21 +346,82 @@ var ERMrest = (function(module) {
+
+                //  get and format the data
+                // var formattedData = formatData(data);
+
+                // construct the uri
+                var uri = this._location.compactUri;
+                for (var i = 0; i < defaults.length; i++) {
+                    uri += (i === 0 ? "?defaults=" : ',') + module._fixedEncodeURIComponent(defaults[i]);
+                }
+
                 //  do the 'post' call
-                //  get the results from post (of course in a promise func)
-                //  make a page of tuples of the results (unless error)
-                //  new page will have a new reference (uri that filters on a disjunction of ids of these tuples)
-                //  resolve the promise, passing back the page
-                notimplemented();
+                module._http.post(uri, data).then(function(response) {
+                    //  new page will have a new reference (uri that filters on a disjunction of ids of these tuples)
+                    var uri = self._location.compactUri + '/',
+                        keyName;
+
+                    self._shortestKey.push(self._table.columns.get("c_int"));
+                    // loop through each returned Row and get the key value
+                    for (var j = 0; j < response.data.length; j++) {
+                        if (j !== 0)
+                            uri += ';';
+                        // shortest key is made up from one column
+                        if (self._shortestKey.length == 1) {
+                            keyName = self._shortestKey[0].name;
+                            uri += keyName + '=' + response.data[j][keyName];
+                        } else {
+                            uri += '(';
+                            for (var k = 0; k < self._shortestKey.length; k++) {
+                                if (k !== 0)
+                                    uri += '&';
+                                keyName = self._shortestKey[k].name;
+                                uri += keyName + '=' + response.data[j][keyName];
+                            }
+                            uri += ')';
+                        }
+                    }
+
+                    var ref = new Reference(module._parse(uri));
+                    //  make a page of tuples of the results (unless error)
+                    var page = new Page(ref, response.data, false, false);
+
+                    //  resolve the promise, passing back the page
+                    return defer.resolve(page);
+                }, function error(response) {
+                    var error = module._responseToError(response);
+                    return defer.reject(error);
+                });
+
+                return defer.promise;
             }
             catch (e) {
                 return module._q.reject(e);
             }
+
+            function getDefaults() {
+                var defaults = module._columnDiff(self._table.columns._columns, self.columns);
```

Comment:
So 'defaults' are defined by the {table columns} - {reference columns}. Is there some heuristic about why? I thought that 'defaults' are something that are in the table specification in ermrest.